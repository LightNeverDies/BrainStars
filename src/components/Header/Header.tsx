import React from 'react'
import style from './header.module.css'

const Header = () => {
    return (
        <div className={style.header}/>
    )
}

export default Header