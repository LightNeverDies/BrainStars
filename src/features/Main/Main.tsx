import React from "react"
import Header from '../../components/Header/Header'
import Movies from "../Movies/Movies"
import Edit from '../Movies/EditMovies'
import Add from '../Movies/AddMovies'

const Main = (props: any) => {

    const bodyRender = () => {
        switch(props.type) {
            case "normal":
                return (
                    <Movies/>
                )
            case "add":
                return (
                    <Add/>
                )
            case "edit":
                return (
                    <Edit/>
                )
        }
    }

    return (
        <>
            <Header/>
            {bodyRender()}
        </>
    )
}

export default Main