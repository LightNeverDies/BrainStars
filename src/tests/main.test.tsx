import 'jsdom-global/register'
import '@testing-library/jest-dom'
import React from "react"
import { mount, shallow } from 'enzyme'

import { Provider } from 'react-redux'
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import fetch from 'jest-fetch-mock'

import reducer, { deleteItem, getItem, clearMessage, 
    searchItem, fetchMovies, addMovie, editMovie, 
    deleteMovie, searchMovie, getMovieForEdit } from '../features/Movies/moviesSlice'
import { BrowserRouter as Router, useNavigate } from 'react-router-dom'

import Main from "../features/Main/Main"
import Header from "../components/Header/Header"
import Add from "../features/Movies/AddMovies"
import Movies from "../features/Movies/Movies"
import Edit from "../features/Movies/EditMovies"

const data = require('./data.json')
const mockStore = configureStore([thunk])

import { changedData, searchData } from './changedData'

describe("Rendering Main", () => {
    const dataStore = {
        error: '',
        data,
        filterData: data,
        loading: false,
        editField: {
            id: '',
            title: '',
            director: '',
            distributor: '',
            imdb_rating: '',
            imdb_votes: ''
        },
        successfulMessage: ''
    }

    const storeData = mockStore({movies: dataStore})

    describe("Render Components without crashing", () => {
        it("render Main component without crashing ", () => {
            shallow(<Main/>)
        })
        it("renders Header component without crashing", () => {
            shallow(<Header/>)
        })
        it("renders Movies component without crashing", () => {
            expect(shallow(<Main type={'normal'}/>))
        })
        it("renders Add component without crashing", () => {
            expect(shallow(<Main type={'add'}/>))
        })
        it("renders Edit component without crashing", () => {
           expect(shallow(<Main type={'edit'}/>))
        })
    })

    describe("moviesSlice Test", () => {

        const initialState = {
            error: '',
            data: [],
            filterData: [],
            loading: false,
            editField: {
                id: '',
                title: '',
                director: '',
                distributor: '',
                imdb_rating: '',
                imdb_votes: ''
            },
            successfulMessage: ''
        }

        const payload = {
            id: 1,
            director: 'None',
            title: 'None',
            distributor: 'None',
            imdb_rating: 2.2,
            imdb_votes: 10000,
        }

        describe("Reducers test", () => {
            it("should return the initial state", () => {
                expect(reducer(undefined, {type: undefined})).toEqual(initialState)
            })
    
            it("should return getItem", () => {
                expect(reducer(undefined, getItem(payload))).toEqual({
                    data: [],
                    editField: payload,
                    error: '',
                    filterData: [],
                    loading: false,
                    successfulMessage: ''
                })
            })
    
            it("should remove item", () => {
    
                const insideState = {
                    error: '',
                    data,
                    filterData: data,
                    loading: false,
                    editField: {
                        id: '',
                        title: '',
                        director: '',
                        distributor: '',
                        imdb_rating: '',
                        imdb_votes: ''
                    },
                    successfulMessage: ''
                }
    
                const insidePayload = data[0]
    
                expect(reducer(insideState, deleteItem(insidePayload))).toEqual(changedData)
            })
    
            it("should clear message", () => {
                const insideState = {
                    error: '',
                    data: [],
                    filterData: [],
                    loading: false,
                    editField: {
                        id: '',
                        title: '',
                        director: '',
                        distributor: '',
                        imdb_rating: '',
                        imdb_votes: ''
                    },
                    successfulMessage: 'Change has been made!'
                }
    
                expect(reducer(insideState, clearMessage())).toEqual(initialState)
            })
    
            it("should search for item", () => {
                const insideState = {
                    error: '',
                    data: [],
                    filterData: data,
                    loading: false,
                    editField: {
                        id: '',
                        title: '',
                        director: '',
                        distributor: '',
                        imdb_rating: '',
                        imdb_votes: ''
                    },
                    successfulMessage: ''
                }
    
                const insidePayload = "1941"
    
                expect(reducer(insideState, searchItem(insidePayload))).toEqual(searchData)
            })
        })

        describe("ExtraReducers test", () => {
            const insideTest = {
                error: '',
                data: [],
                filterData: [],
                loading: false,
                editField: {
                    id: '',
                    title: '',
                    director: '',
                    distributor: '',
                    imdb_rating: '',
                    imdb_votes: ''
                },
                successfulMessage: ''
            }

            it('should fetch action pending', () => {

                const expectedResult = {
                    error: '',
                    data: [],
                    filterData: [],
                    loading: true,
                    editField: {
                        id: '',
                        title: '',
                        director: '',
                        distributor: '',
                        imdb_rating: '',
                        imdb_votes: ''
                    },
                    successfulMessage: ''
                }

                expect(reducer(insideTest, { type: fetchMovies.pending})).toEqual(expectedResult)
            })

            it('should fetch action fulfilled - empty data', async() => {
                const expectedResult = {
                    error: '',
                    data: undefined,
                    filterData: undefined,
                    loading: false,
                    editField: {
                        id: '',
                        title: '',
                        director: '',
                        distributor: '',
                        imdb_rating: '',
                        imdb_votes: ''
                    },
                    successfulMessage: ''
                }

                expect(reducer(insideTest, { type: fetchMovies.fulfilled})).toEqual(expectedResult)
            })

            it('should fetch action rejected', () => {
                const expectedResult = {
                    error: undefined,
                    data: [],
                    filterData: [],
                    loading: false,
                    editField: {
                        id: '',
                        title: '',
                        director: '',
                        distributor: '',
                        imdb_rating: '',
                        imdb_votes: ''
                    },
                    successfulMessage: ''
                }

                expect(reducer(insideTest, { type: fetchMovies.rejected})).toEqual(expectedResult)
            })

            it('should addMovie action pending', () => {
                const expectedResult = {
                    error: "",
                    data: [],
                    filterData: [],
                    loading: true,
                    editField: {
                        id: '',
                        title: '',
                        director: '',
                        distributor: '',
                        imdb_rating: '',
                        imdb_votes: ''
                    },
                    successfulMessage: "New record has been added!"
                }
                expect(reducer(insideTest, { type: addMovie.pending})).toEqual(expectedResult)
            })
            
            it('should addMovie action rejected', () => {
                const expectedResult = {
                    error: "Something went wrong!",
                    data: [],
                    filterData: [],
                    loading: false,
                    editField: {
                        id: '',
                        title: '',
                        director: '',
                        distributor: '',
                        imdb_rating: '',
                        imdb_votes: ''
                    },
                    successfulMessage: ''
                }
                expect(reducer(insideTest, { type: addMovie.rejected})).toEqual(expectedResult)
            })

            it('should editMovie action pending', () => {
                const expectedResult = {
                    error: "",
                    data: [],
                    filterData: [],
                    loading: true,
                    editField: {
                        id: '',
                        title: '',
                        director: '',
                        distributor: '',
                        imdb_rating: '',
                        imdb_votes: ''
                    },
                    successfulMessage: "Record has been changed!"
                }

                expect(reducer(insideTest, { type: editMovie.pending})).toEqual(expectedResult)
            })

            it('should editMovie action rejected', () => {
                const expectedResult = {
                    error: "Something went wrong!",
                    data: [],
                    filterData: [],
                    loading: false,
                    editField: {
                        id: '',
                        title: '',
                        director: '',
                        distributor: '',
                        imdb_rating: '',
                        imdb_votes: ''
                    },
                    successfulMessage: ""
                }

                expect(reducer(insideTest, { type: editMovie.rejected})).toEqual(expectedResult)
            })
        })

        describe("Actions test", () => {
            const insidePayload = {
                id: 1,
                director: 'None',
                title: 'None',
                distributor: 'None',
                imdb_rating: 2.2,
                imdb_votes: 10000,
            }

            const expectedResult = {
                "id": 4,
                "title": 1941,
                "director": "Steven Spielberg",
                "distributor": "Universal",
                "imdb_rating": 5.6,
                "imdb_votes": 13364
            }
    

            it('should fetchMovies - resolve', async() => {
                fetch.mockResponseOnce(JSON.stringify(data))
                const result = await storeData.dispatch(fetchMovies() as any)
                expect(result.payload).toEqual(data)
            })

            it('should fetchMovies - rejected', async() => {
                fetch.mockRejectOnce(new Error("Something went wrong"))
                const result = await storeData.dispatch(fetchMovies() as any)
                expect(result.payload).toEqual(new Error("Something went wrong"))
            })

            it('should addMovie - correct', async() => {
                const result = await storeData.dispatch(addMovie(insidePayload) as any)
                expect(result.payload).toEqual(insidePayload)
            })

            it('should editMovie - correct', async() => {
                const result = await storeData.dispatch(editMovie(insidePayload) as any)
                expect(result.payload).toEqual(insidePayload)
            })

            it('should deleteMovie - correct', async() => {
                const result = await storeData.dispatch(deleteMovie(expectedResult) as any)
                expect(result.payload).toEqual(expectedResult)
            })

            it('should editMovie - correct', async() => {
                const result = await storeData.dispatch(searchMovie(expectedResult) as any)
                expect(result.payload).toEqual(expectedResult)
            })

            it('should getMovieForEdit - correct', async() => {
                const result = await storeData.dispatch(getMovieForEdit(expectedResult) as any)
                expect(result.payload).toEqual(expectedResult)
            })
        })

    })

    describe("Render Movie Component", () => {
        describe('Render Component', () => {
            let wrapper: any
            beforeEach(() => {
                wrapper = render(
                    <Provider store={storeData}>
                        <Router>
                            <Movies/>
                        </Router>
                    </Provider>
                )
            })
    
            it('render component', () => {
                const { asFragment } = wrapper
                expect(asFragment()).toMatchSnapshot()
            })
        })
       
        describe('Function in component' , () => {
            let mockedFunction: any, wrapper: any
            beforeEach(() => {
                mockedFunction = jest.fn();
                wrapper = mount
                    ( 
                        <Provider store={storeData}>
                            <Router>
                                <Movies item={mockedFunction}/>
                            </Router>
                        </Provider>
                    )
            })


            it('test Delete Button', () => {
                wrapper.find('#delete').at(0).simulate('click')
                expect(mockedFunction.mock.calls.length).toEqual(0)
            })

            it('test Table Movies', () => {
                const table = wrapper.find('table')
                expect(table).toHaveLength(1)
            })

            it('test Search Input field', () => {
                const onSearchMock = jest.fn();
                wrapper.find("#search").simulate('change')
                expect(onSearchMock).toBeCalledTimes(0)
            })
        })



    })

    describe("Render AddMovies Component", () => {
        it('render component', () => {
            const wrapper = render(
                <Provider store={storeData}>
                    <Router>
                        <Add/>
                    </Router>
                </Provider>
            )
            const { asFragment } = wrapper
            expect(asFragment()).toMatchSnapshot()
        })
    })

    describe("Render EditMovies Component", () => {
        it('render component', () => {
            const wrapper = render(
                <Provider store={storeData}>
                    <Router>
                        <Edit/>
                    </Router>
                </Provider>
            )
            const { asFragment } = wrapper
            expect(asFragment()).toMatchSnapshot()
        })
    })


})
