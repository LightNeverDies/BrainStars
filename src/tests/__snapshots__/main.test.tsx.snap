// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Main Render AddMovies Component render component 1`] = `
<DocumentFragment>
  <div>
    <form>
      <h1>
        Create Movie
      </h1>
      <div>
        <label>
          Id:
        </label>
        <input
          name="id"
          type="text"
        />
      </div>
      <div>
        <label>
          Title:
        </label>
        <input
          name="title"
          type="text"
        />
      </div>
      <div>
        <label>
          Director:
        </label>
        <input
          name="director"
          type="text"
        />
      </div>
      <div>
        <label>
          Distributor:
        </label>
        <input
          name="distributor"
          type="text"
        />
      </div>
      <div>
        <label>
          Imdb_rating:
        </label>
        <input
          name="imdb_rating"
          type="text"
        />
      </div>
      <div>
        <label>
          Imdb_votes:
        </label>
        <input
          name="imdb_votes"
          type="text"
        />
      </div>
      <div>
        <input
          id="back"
          type="submit"
          value="Save movie"
        />
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Rendering Main Render EditMovies Component render component 1`] = `
<DocumentFragment>
  <div>
    <form>
      <h1>
        Edit Movie with ID 
      </h1>
      <div>
        <label>
          Id:
        </label>
        <input
          name="id"
        />
      </div>
      <div>
        <label>
          Title:
        </label>
        <input
          name="title"
          type="text"
        />
      </div>
      <div>
        <label>
          Director:
        </label>
        <input
          name="director"
          type="text"
        />
      </div>
      <div>
        <label>
          Distributor:
        </label>
        <input
          name="distributor"
          type="text"
        />
      </div>
      <div>
        <label>
          Imdb_rating:
        </label>
        <input
          name="imdb_rating"
          type="text"
        />
      </div>
      <div>
        <label>
          Imdb_votes:
        </label>
        <input
          name="imdb_votes"
          type="text"
        />
      </div>
      <div>
        <input
          type="submit"
          value="Update movie"
        />
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Rendering Main Render Movie Component Render Component render component 1`] = `
<DocumentFragment>
  <div>
    <p />
    <div>
      <div />
      <div>
        <div>
          <input
            id="search"
            placeholder="Search"
          />
        </div>
        <div>
          <h1>
            Movies
          </h1>
          <button
            id="add"
          >
            Add
          </button>
        </div>
      </div>
      <div />
    </div>
    <div>
      <table>
        <thead>
          <tr>
            <th>
              ID
            </th>
            <th>
              TITLE
            </th>
            <th>
              DIRECTOR
            </th>
            <th>
              DISTRIBUTOR
            </th>
            <th>
              IMDB_RATING
            </th>
            <th>
              IMDB_VOTES
            </th>
            <th>
              ACTIONS
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td>
              Following
            </td>
            <td>
              Christopher Nolan
            </td>
            <td>
              Zeitgeist
            </td>
            <td>
              7.7
            </td>
            <td>
              15133
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              2
            </td>
            <td>
              12 Angry Men
            </td>
            <td>
              Sidney Lumet
            </td>
            <td>
              United Artists
            </td>
            <td>
              8.9
            </td>
            <td>
              119101
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              3
            </td>
            <td>
              Twelve Monkeys
            </td>
            <td>
              Terry Gilliam
            </td>
            <td>
              Universal
            </td>
            <td>
              8.1
            </td>
            <td>
              169858
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              4
            </td>
            <td>
              1941
            </td>
            <td>
              Steven Spielberg
            </td>
            <td>
              Universal
            </td>
            <td>
              5.6
            </td>
            <td>
              13364
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              5
            </td>
            <td>
              24 7: Twenty Four Seven
            </td>
            <td>
              Shane Meadows
            </td>
            <td>
              October Films
            </td>
            <td>
              6.9
            </td>
            <td>
              1417
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              6
            </td>
            <td>
              Twin Falls Idaho
            </td>
            <td>
              Michael Polish
            </td>
            <td>
              Sony Pictures Classics
            </td>
            <td>
              7.1
            </td>
            <td>
              2810
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              7
            </td>
            <td>
              3 Men and a Baby
            </td>
            <td>
              Leonard Nimoy
            </td>
            <td>
              Walt Disney Pictures
            </td>
            <td>
              5.7
            </td>
            <td>
              16764
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              8
            </td>
            <td>
              Four Rooms
            </td>
            <td>
              Robert Rodriguez
            </td>
            <td>
              Miramax
            </td>
            <td>
              6.4
            </td>
            <td>
              34328
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              9
            </td>
            <td>
              The Four Seasons
            </td>
            <td>
              Alan Alda
            </td>
            <td>
              Universal
            </td>
            <td>
              7
            </td>
            <td>
              1814
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              10
            </td>
            <td>
              Four Weddings and a Funeral
            </td>
            <td>
              Mike Newell
            </td>
            <td>
              Gramercy
            </td>
            <td>
              7.1
            </td>
            <td>
              39003
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              11
            </td>
            <td>
              Nine 1/2 Weeks
            </td>
            <td>
              Adrian Lyne
            </td>
            <td>
              MGM
            </td>
            <td>
              5.4
            </td>
            <td>
              12731
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              12
            </td>
            <td>
              The Abyss
            </td>
            <td>
              James Cameron
            </td>
            <td>
              20th Century Fox
            </td>
            <td>
              7.6
            </td>
            <td>
              51018
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              13
            </td>
            <td>
              Ace Ventura: Pet Detective
            </td>
            <td>
              Tom Shadyac
            </td>
            <td>
              Warner Bros.
            </td>
            <td>
              6.6
            </td>
            <td>
              63543
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              14
            </td>
            <td>
              Ace Ventura: When Nature Calls
            </td>
            <td>
              Steve Oedekerk
            </td>
            <td>
              Warner Bros.
            </td>
            <td>
              5.6
            </td>
            <td>
              51275
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              15
            </td>
            <td>
              The Alamo
            </td>
            <td>
              John Wayne
            </td>
            <td>
              United Artists
            </td>
            <td>
              5.9
            </td>
            <td>
              10063
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              16
            </td>
            <td>
              Alive
            </td>
            <td>
              Frank Marshall
            </td>
            <td>
              Walt Disney Pictures
            </td>
            <td>
              3.2
            </td>
            <td>
              124
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              17
            </td>
            <td>
              Amen
            </td>
            <td>
              Costa-Gavras
            </td>
            <td>
              Kino International
            </td>
            <td>
              7.4
            </td>
            <td>
              5416
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              18
            </td>
            <td>
              American Graffiti
            </td>
            <td>
              George Lucas
            </td>
            <td>
              Universal
            </td>
            <td>
              7.6
            </td>
            <td>
              30952
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              19
            </td>
            <td>
              The American President
            </td>
            <td>
              Rob Reiner
            </td>
            <td>
              Sony Pictures
            </td>
            <td>
              6.8
            </td>
            <td>
              22780
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              20
            </td>
            <td>
              Annie Hall
            </td>
            <td>
              Woody Allen
            </td>
            <td>
              MGM
            </td>
            <td>
              8.2
            </td>
            <td>
              65406
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              21
            </td>
            <td>
              The Adventures of Huck Finn
            </td>
            <td>
              Stephen Sommers
            </td>
            <td>
              Walt Disney Pictures
            </td>
            <td>
              5.8
            </td>
            <td>
              3095
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              22
            </td>
            <td>
              The Apartment
            </td>
            <td>
              Billy Wilder
            </td>
            <td>
              United Artists
            </td>
            <td>
              8.4
            </td>
            <td>
              36485
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              23
            </td>
            <td>
              Apocalypse Now
            </td>
            <td>
              Francis Ford Coppola
            </td>
            <td>
              MGM
            </td>
            <td>
              8.6
            </td>
            <td>
              173141
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              24
            </td>
            <td>
              Arachnophobia
            </td>
            <td>
              Frank Marshall
            </td>
            <td>
              Walt Disney Pictures
            </td>
            <td>
              6.2
            </td>
            <td>
              20528
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              25
            </td>
            <td>
              Barry Lyndon
            </td>
            <td>
              Stanley Kubrick
            </td>
            <td>
              Warner Bros.
            </td>
            <td>
              8.1
            </td>
            <td>
              39909
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              26
            </td>
            <td>
              Babe
            </td>
            <td>
              Chris Noonan
            </td>
            <td>
              Universal
            </td>
            <td>
              7.3
            </td>
            <td>
              35644
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              27
            </td>
            <td>
              Baby's Day Out
            </td>
            <td>
              Patrick Read Johnson
            </td>
            <td>
              20th Century Fox
            </td>
            <td>
              5
            </td>
            <td>
              8332
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              28
            </td>
            <td>
              Bound by Honor
            </td>
            <td>
              Taylor Hackford
            </td>
            <td>
              Walt Disney Pictures
            </td>
            <td>
              7.7
            </td>
            <td>
              10142
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              29
            </td>
            <td>
              Back to the Future
            </td>
            <td>
              Robert Zemeckis
            </td>
            <td>
              Universal
            </td>
            <td>
              8.4
            </td>
            <td>
              201598
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              30
            </td>
            <td>
              Back to the Future Part II
            </td>
            <td>
              Robert Zemeckis
            </td>
            <td>
              Universal
            </td>
            <td>
              7.5
            </td>
            <td>
              87341
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              31
            </td>
            <td>
              Back to the Future Part III
            </td>
            <td>
              Robert Zemeckis
            </td>
            <td>
              Universal
            </td>
            <td>
              7.1
            </td>
            <td>
              77541
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              32
            </td>
            <td>
              Butch Cassidy and the Sundance Kid
            </td>
            <td>
              George Roy Hill
            </td>
            <td>
              20th Century Fox
            </td>
            <td>
              8.2
            </td>
            <td>
              57602
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              33
            </td>
            <td>
              Bad Boys
            </td>
            <td>
              Michael Bay
            </td>
            <td>
              Sony Pictures
            </td>
            <td>
              6.6
            </td>
            <td>
              53929
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              34
            </td>
            <td>
              Body Double
            </td>
            <td>
              Brian De Palma
            </td>
            <td>
              Sony/Columbia
            </td>
            <td>
              6.4
            </td>
            <td>
              9738
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              35
            </td>
            <td>
              Ben-Hur
            </td>
            <td>
              William Wyler
            </td>
            <td>
              MGM
            </td>
            <td>
              8.2
            </td>
            <td>
              58510
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              36
            </td>
            <td>
              Before Sunrise
            </td>
            <td>
              Richard Linklater
            </td>
            <td>
              Sony Pictures
            </td>
            <td>
              8
            </td>
            <td>
              39705
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              37
            </td>
            <td>
              Beauty and the Beast
            </td>
            <td>
              Gary Trousdale
            </td>
            <td>
              Walt Disney Pictures
            </td>
            <td>
              3.4
            </td>
            <td>
              354
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              38
            </td>
            <td>
              The Best Years of Our Lives
            </td>
            <td>
              William Wyler
            </td>
            <td>
              RKO Radio Pictures
            </td>
            <td>
              8.2
            </td>
            <td>
              17338
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              39
            </td>
            <td>
              Battle for the Planet of the Apes
            </td>
            <td>
              Jack Lee Thompson
            </td>
            <td>
              20th Century Fox
            </td>
            <td>
              5
            </td>
            <td>
              6094
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              40
            </td>
            <td>
              Bogus
            </td>
            <td>
              Norman Jewison
            </td>
            <td>
              Warner Bros.
            </td>
            <td>
              4.8
            </td>
            <td>
              2742
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              41
            </td>
            <td>
              Beverly Hills Cop
            </td>
            <td>
              Martin Brest
            </td>
            <td>
              Paramount Pictures
            </td>
            <td>
              7.3
            </td>
            <td>
              45065
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              42
            </td>
            <td>
              Beverly Hills Cop II
            </td>
            <td>
              Tony Scott
            </td>
            <td>
              Paramount Pictures
            </td>
            <td>
              6.1
            </td>
            <td>
              29712
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              43
            </td>
            <td>
              Beverly Hills Cop III
            </td>
            <td>
              John Landis
            </td>
            <td>
              Paramount Pictures
            </td>
            <td>
              5
            </td>
            <td>
              21199
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              44
            </td>
            <td>
              Big
            </td>
            <td>
              Penny Marshall
            </td>
            <td>
              20th Century Fox
            </td>
            <td>
              7.2
            </td>
            <td>
              49256
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
          <tr>
            <td>
              45
            </td>
            <td>
              The Big Parade
            </td>
            <td>
              King Vidor
            </td>
            <td>
              MGM
            </td>
            <td>
              8.4
            </td>
            <td>
              2600
            </td>
            <td>
              <button
                id="edit"
              >
                Edit
              </button>
              <button
                id="delete"
              >
                Delete
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
